src/experimentation-opportunities/experimentation-opportunities.js

```diff
@@ -11,17 +11,251 @@
+  const highOrganicLowCtrOpportunities = opportunites.filter((oppty) => oppty.type === 'high-organic-low-ctr')
+    .map((oppty) => {
+      const { pageViews, trackedPageKPIValue, trackedKPISiteAverage } = oppty;
+      const opportunityImpact = Math.floor(pageViews
+          * (trackedKPISiteAverage - CTR_THRESHOLD_MARGIN - trackedPageKPIValue));
+      return {
+        ...oppty,
+        opportunityImpact,
+      };
+    });
+  log.info(`Found ${highOrganicLowCtrOpportunities.length} high organic low CTR opportunities`);
+  highOrganicLowCtrOpportunities.sort((a, b) => b.opportunityImpact - a.opportunityImpact);
+  const topHighOrganicLowCtrOpportunities = highOrganicLowCtrOpportunities.slice(
+    0,
+    MAX_OPPORTUNITIES,
+  );
+  const topHighOrganicUrls = topHighOrganicLowCtrOpportunities.map((oppty) => ({
+    url: oppty.page,
+  }));
+  log.info(`Triggering scrape for [${JSON.stringify(topHighOrganicUrls, null, 2)}]`);
+  await sqs.sendMessage(process.env.SCRAPING_JOBS_QUEUE_URL, {
+    processingType: 'default',
+    jobId: site.getId(),
+    urls: topHighOrganicUrls,
+  });
+  // wait for the scrape to complete
+  // TODO: replace this with a SQS message to run another handler to process the scraped content,
+  // to eliminate duplicate lambda run time
+
+  // generate the guidance for the top opportunities in parallel
+  const promises = topHighOrganicLowCtrOpportunities.map(
+    (oppty) => updateRecommendations(oppty, context, site),
+  );
+  await Promise.all(promises);
+  for (const oppty of topHighOrganicLowCtrOpportunities) {
+    // eslint-disable-next-line no-await-in-loop
+    // await updateRecommendations(oppty, context, site);
+    // update the oppty in the opporrtunities list
+    const index = opportunites.findIndex(
+      (opp) => opp.page === oppty.page && opp.type === oppty.type,
+    );
+    if (index !== -1) {
+      // eslint-disable-next-line no-param-reassign
+      opportunites[index] = oppty;
+    }
+  }
+}
+
+async function createOrUpdateOpportunityEntity(opportunity, context, existingOpportunities) {
+  const { log, dataAccess } = context;
+  const { Opportunity } = dataAccess;
+  const existingOpportunity = existingOpportunities.find(
+    (oppty) => (oppty.getType() === opportunity.type) && oppty.getData()
+    && (oppty.getData().page === opportunity.data.page),
+  );
+  if (existingOpportunity) {
+    if (existingOpportunity.getStatus() === 'NEW') {
+      // remove and create a new opportunity entity with new data
+      log.info(`[${opportunity.type}] Opportunity entity with status: NEW for ${opportunity.data.page} exists, so removing it and creating a new opportunity entity`);
+      await existingOpportunity.remove();
+    } else {
+      log.info(`[${opportunity.type}] Opportunity entity already exists with status: ${existingOpportunity.getStatus()} for ${opportunity.data.page}, so skipping`);
+      return false;
+    }
+  }
+  await Opportunity.create(opportunity);
```

Comment:
does this require error handling, as done here:
https://github.com/adobe/spacecat-audit-worker/pull/475/files#diff-212dd58956c947938959c789cc3912af8f1b884de5543486166247e12f8ef215R188

Reply:
handled this in the caller function, so we don't need additional handling